docs/c-api.rst

```diff
@@ -682,9 +682,121 @@ File format errors
 .. doxygengroup:: FILE_FORMAT_ERROR_GROUP
         :content-only:
 
+--------------------
+Out-of-bounds errors
+--------------------
+
+.. doxygengroup:: OOB_ERROR_GROUP
+        :content-only:
+
+-----------
+Edge errors
+-----------
+
+.. doxygengroup:: EDGE_ERROR_GROUP
+        :content-only:
+
 
-.. todo:: Add in groups for rest of the error types and document.
+-----------
+Site errors
+-----------
+
+.. doxygengroup:: SITE_ERROR_GROUP
+        :content-only:
+
+
+---------------
+Mutation errors
+---------------
+
+.. doxygengroup:: MUTATION_ERROR_GROUP
+        :content-only:
+
+
+----------------
+Migration errors
+----------------
+
+.. doxygengroup:: MIGRATION_ERROR_GROUP
+        :content-only:
+
+-------------
+Sample errors
+-------------
+
+.. doxygengroup:: SAMPLE_ERROR_GROUP
+        :content-only:
+
+------------
+Table errors
+------------
+
+.. doxygengroup:: TABLE_ERROR_GROUP
+        :content-only:
+
+------------
+Stats errors
+------------
 
+.. doxygengroup:: STATS_ERROR_GROUP
+        :content-only:
+
+-----------------------
+Mutation mapping errors
+-----------------------
+
+.. doxygengroup:: MAPPING_ERROR_GROUP
+        :content-only:
+
+------------------------
+Genotype decoding errors
+------------------------
+
+.. doxygengroup:: GENOTYPE_ERROR_GROUP
+        :content-only:
+
+----------------------
+Distance metric errors
+----------------------
+
+.. doxygengroup:: DISTANCE_ERROR_GROUP
+        :content-only:
+
+-------------------------
```

Comment:
Maybe we should leave HAPLOTYPE_ERROR_GROUP,  DISTANCE_ERROR_GROUP, IBD_ERROR_GROUP, MAPPING_ERROR_GROUP, and STATS_ERROR_GROUP out for now? Seems odd to put these in to the end-user documentation when we're not providing the corresponding functions in the API. You've done the hard work, to adding the groups back in to the RST here will be very easy later.

Reply:
Yeah, got a bit carried away. Will fix in the in-progress polish PR.